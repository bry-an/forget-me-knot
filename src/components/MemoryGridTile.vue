<template>
  <div>
    {{ tile.clicked }}
    <img
      class="cursor-pointer"
      @click="$emit('select-tile', tile, index)"
      :src="tile.slug"
      alt="memory-game-tile"
    />
  </div>
</template>

<script>
export default {
  name: "MemoryGridTile",
  props: {
    tile: {
      type: Object,
      default: () => ({}),
    },
    index: {
      type: Number,
      default: 0,
    },
  },
};
</script>
<style lang="scss" scoped></style>
