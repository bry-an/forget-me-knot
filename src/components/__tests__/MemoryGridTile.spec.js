import MemoryGridTile from "../MemoryGridTile.vue";
import { componentMountFactory } from "../../test/componentTestFactory.js";

const visibleSampleTile = {
  slug:
    "https://media0.giphy.com/media/3otPotnGlA5FSF2fKw/200.gif?cid=efc6841ejd0hn2k52nkal7r914iiteuuttvc571t7krnx9ye&rid=200.gif&ct=g",
  key:
    "https://media0.giphy.com/media/3otPotnGlA5FSF2fKw/200.gif?cid=efc6841ejd0hn2k52nkal7r914iiteuuttvc571t7krnx9ye&rid=200.gif&ct=g-2",
  clicked: false,
  display: true,
  sibling: {
    slug:
      "https://media0.giphy.com/media/3otPotnGlA5FSF2fKw/200.gif?cid=efc6841ejd0hn2k52nkal7r914iiteuuttvc571t7krnx9ye&rid=200.gif&ct=g",
    key:
      "https://media0.giphy.com/media/3otPotnGlA5FSF2fKw/200.gif?cid=efc6841ejd0hn2k52nkal7r914iiteuuttvc571t7krnx9ye&rid=200.gif&ct=g-1",
    clicked: false,
    display: true,
  },
};

const nonvisibleSampleTile = {
  ...visibleSampleTile,
  display: false,
};

const buildComponentWrapper = componentMountFactory(MemoryGridTile);

describe("MemoryGridTile", () => {
  it("Mounts", () => {
    const wrapper = buildComponentWrapper();
    expect(wrapper.exists()).toBeTruthy();
  });
  it("Correctly loads the tile's image in the card's underside", () => {
    const wrapper = buildComponentWrapper({
      propsData: {
        tile: visibleSampleTile,
      },
    });
    expect(wrapper.find("[data-test=image-under-tile]").attributes().src).toBe(
      visibleSampleTile.slug
    );
  });
  it("Does not show tile if visible set to false", () => {
    const wrapper = buildComponentWrapper({
      propsData: {
        tile: nonvisibleSampleTile,
      },
    });
    expect(wrapper.find("[data-test=flip-tile]").classes().visible).toBeFalsy();
  });
  it("Matches visible snapshot", () => {
    const wrapper = buildComponentWrapper({
      propsData: {
        tile: visibleSampleTile,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("Matches nonvisible snapshot", () => {
    const wrapper = buildComponentWrapper({
      propsData: {
        tile: nonvisibleSampleTile,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
