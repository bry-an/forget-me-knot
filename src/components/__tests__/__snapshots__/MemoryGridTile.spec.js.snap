// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemoryGridTile Matches nonvisible snapshot 1`] = `
<div
  class="relative invisible"
>
  <!--v-if-->
  <div
    class="flip-tile cursor-pointer"
    data-test="flip-tile"
  >
    <div
      class="flip-tile-inner"
    >
      <div
        class="flip-tile-front"
      >
        <img
          alt="memory-card-front"
          class="rounded-md"
          src=""
        />
      </div>
      <div
        class="flip-tile-back flex items-center rounded-md"
      >
        <img
          alt="memory-game-tile"
          class="rounded-md"
          data-test="image-under-tile"
          src="https://media0.giphy.com/media/3otPotnGlA5FSF2fKw/200.gif?cid=efc6841ejd0hn2k52nkal7r914iiteuuttvc571t7krnx9ye&rid=200.gif&ct=g"
          style="background: url(https://media0.giphy.com/media/3otPotnGlA5FSF2fKw/200.gif?cid=efc6841ejd0hn2k52nkal7r914iiteuuttvc571t7krnx9ye&rid=200.gif&ct=g);"
          width="150"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`MemoryGridTile Matches visible snapshot 1`] = `
<div
  class="relative visible"
>
  <!--v-if-->
  <div
    class="flip-tile cursor-pointer"
    data-test="flip-tile"
  >
    <div
      class="flip-tile-inner"
    >
      <div
        class="flip-tile-front"
      >
        <img
          alt="memory-card-front"
          class="rounded-md"
          src=""
        />
      </div>
      <div
        class="flip-tile-back flex items-center rounded-md"
      >
        <img
          alt="memory-game-tile"
          class="rounded-md"
          data-test="image-under-tile"
          src="https://media0.giphy.com/media/3otPotnGlA5FSF2fKw/200.gif?cid=efc6841ejd0hn2k52nkal7r914iiteuuttvc571t7krnx9ye&rid=200.gif&ct=g"
          style="background: url(https://media0.giphy.com/media/3otPotnGlA5FSF2fKw/200.gif?cid=efc6841ejd0hn2k52nkal7r914iiteuuttvc571t7krnx9ye&rid=200.gif&ct=g);"
          width="150"
        />
      </div>
    </div>
  </div>
</div>
`;
